chain/src/txhashset.rs

```diff
@@ -1224,27 +1224,45 @@ fn input_pos_to_rewind(
 			head_header.height,
 			block_header.height
 		);
-		return Ok(bitmap);
+		return Ok(Bitmap::create());
 	}
 
-	//
-	// TODO - rework this loop to use Bitmap::fast_or() on a vec of bitmaps.
-	//
-	loop {
-		if current == block_header.hash() {
-			break;
+	let bitmap_fast_or = |b_res, block_input_bitmaps: &mut Vec<Bitmap>| -> Option<Bitmap> {
+		if let Some(b) = b_res {
+			block_input_bitmaps.push(b);
+			if block_input_bitmaps.len() < 256 {
```

Comment:
Do we really need this additional complexity for short rewinds?
Might be easier to maintain (and understand) if we just did the `fast_or` for every rewind.

Reply:
thanks for review.
Do you mean push all the bitmaps into the vec[]? then do `fast_or` only once?
For short rewinds, it's Ok. But if the long rewinds (for example the API: http://127.0.0.1:13413/v1/blocks/40000), the memory consumption could be huge, that's why I limit the vec[] size to 256.

Any idea to improve it?

Reply:
I think I was misreading the code here. Thought we were only doing `fast_or()` if we had > 256 bitmaps.
But I _think_ what you're doing is batching them up and running `fast_or()` on every batch of 256 bitmaps so we don't need to maintain a huge vec of bitmaps.

:+1: if that's the case.
But maybe add a comment explaining that?

Reply:
Yes, that's the purpose.
Will add the code comment for that.